### PR TITLE
small fixes related to degredation and CWD/disturbance fluxes

### PR DESCRIPTION
**Pull request by rgknox**

### CODE REVIEW

**Comment by rgknox** [comment]
@ckoven could you take a look at this PR, some changes were needed relevant to degredation.  Particularly changes to impact mortality rates and also the litter fluxes for those.

These changes enable 60 year f10 simulations to complete with logging every year on day 30.

**Comment by rgknox** [comment]
test on cheyenne: /gpfs/fs1/scratch/rgknox/fates-clm-tests/logtime-f10_f10_mg37/run

**Comment by ckoven** [comment]
@rgknox so these fix the mass balance error?

**Comment by rgknox** [comment]
yes, my tests now all pass

**Comment by rgknox** [comment]
@ckoven , i made some comments stating that there is really only an effect when tracking degredation in the canopy layer, as that is the only layer that effects disturbance area.  There is no point in tracking degredation rates in the understory, would you agree with that?

**Comment by huangmy** [comment]
@rgknox I just had a chance to check this. I actually did not see any mass balance errors in my forest clearing simulation with a single wood harvest event at the beginning of the simulation. Is it expected?

**Comment by rgknox** [comment]
@huangmy , my BCI test did not trigger the mass balance error either.  But I did trigger mass balance errors on the global grid.  This mass balance error would only show up if there was appreciable mortality resulting from impact mortality on understory.

**Comment by rgknox** [comment]
@huangmy, do the fixes make sense to you?  The key change is that calculation of the mortality rate for impacts on understory plants, and also making sure that same rate is applied to the generation of litter fluxes.

Sorry I added a bunch of line indent changes as well, which makes it look like more is changed that really happened.

File: biogeochem/EDLoggingMortalityMod.F90 [hunk 1]
```diff
@@ -318,10 +321,19 @@ subroutine logging_litter_fluxes(currentSite, currentPatch, newPatch, patch_site
...
+            ! handled in EDPhysiology.  Disturbance generating mortality
+            ! are those cohorts in the top canopy layer, or those
+            ! plants that were impacted. Thus, no direct dead can occur
+            ! here, and indirect are impacts.
+
             if(EDPftvarcon_inst%woody(currentCohort%pft) == 1)then
                direct_dead   = 0.0_r8
-               indirect_dead = logging_coll_under_frac * currentCohort%n * &
-                     (patch_site_areadis/currentPatch%area)  !kgC/site/day
+               indirect_dead = logging_coll_under_frac * &
```
Comment (rgknox) [inline biogeochem/EDLoggingMortalityMod.F90]: key change 1

File: biogeochem/EDPatchDynamicsMod.F90 [hunk 2]
```diff
@@ -759,7 +759,7 @@ subroutine spawn_patches( currentSite, bc_in)
                          ! LOGGING SURVIVORSHIP OF UNDERSTORY PLANTS IS SET AS A NEW PARAMETER 
                          ! in the fatesparameter files 
                          nc%n = nc%n * (1.0_r8 - &
-                               currentPatch%fract_ldist_not_harvested * logging_coll_under_frac)
+                              (1.0_r8-currentPatch%fract_ldist_not_harvested) * logging_coll_under_frac)
```
Comment (rgknox) [inline biogeochem/EDPatchDynamicsMod.F90]: key change 2

File: biogeochem/EDPhysiologyMod.F90 [hunk 3]
```diff
@@ -1168,137 +1168,136 @@ subroutine CWD_Input( currentSite, currentPatch)
               (1.0_r8-EDPftvarcon_inst%allom_agb_frac(currentCohort%pft))
       enddo
```
Comment (rgknox) [inline biogeochem/EDPhysiologyMod.F90]: these are just line indent changes

**Comment by rgknox** [comment]
The changes in EDPhysiologyMod are all just line indent fixes

File: biogeochem/EDPatchDynamicsMod.F90 (hunk 2 above)
Comment (huangmy) [reply biogeochem/EDPatchDynamicsMod.F90]: So the changes in EDphysiology are mainly for formatting, right? The changes in EDLoggingMortality and EDPatchDynamics look reasonable to me.

Comment (rgknox) [reply biogeochem/EDPatchDynamicsMod.F90]: tha'ts right

Comment (huangmy) [reply biogeochem/EDPatchDynamicsMod.F90]: Sounds good to me. Let me know if I should merge after Charlie reviews it.

**Comment by ckoven** [comment]
I think I approve.  But I feel like I need to really think about this for a while to be completely sure...

**Comment by huangmy** [comment]
@ckoven @rgknox shall I wait then?

**Comment by ckoven** [comment]
no, you should merge it.

**Pull request merged**